src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocument.cs

```diff
@@ -219,6 +222,96 @@ private void RestructureItem(TocItemViewModel item, List<TocItemViewModel> items
             }
         }
 
+        private void ReportDependency(ImmutableList<FileModel> models, IHostService host, ImmutableDictionary<string, TocItemInfo> tocModelCache, int parallelism)
+        {
+            var nearest = new Dictionary<string, Toc>();
+            models.RunAll(model =>
+            {
+                var wrapper = tocModelCache[model.OriginalFileAndType.FullPath];
+
+                // If the TOC file is referenced by other TOC, not report dependency
+                if (wrapper.IsReferenceToc)
+                {
+                    return;
+                }
+                var outputFile = GetOutputPath(model.FileAndType);
+                var item = wrapper.Content;
+                UpdateNearestToc(host, item, model, nearest);
+            },
+            parallelism);
+            foreach (var item in nearest)
+            {
+                host.ReportDependencyFrom(item.Value.Model, item.Key, DependencyTypeName.Metadata);
+            }
+        }
+
+        private void UpdateNearestToc(IHostService host, TocItemViewModel item, FileModel toc, Dictionary<string, Toc> nearest)
+        {
+            var tocHref = item.TocHref;
+            var type = Utility.GetHrefType(tocHref);
+            if (type == HrefType.MarkdownTocFile || type == HrefType.YamlTocFile)
+            {
+                UpdateNearestTocCore(host, tocHref, toc, nearest);
+            }
+            else
+            {
+                if (Utility.IsSupportedRelativeHref(item.Href))
+                {
+                    UpdateNearestTocCore(host, item.Href, toc, nearest);
+                }
+            }
+
+            if (item.Items != null && item.Items.Count > 0)
+            {
+                foreach (var i in item.Items)
+                {
+                    UpdateNearestToc(host, i, toc, nearest);
+                }
+            }
+        }
+
+        private void UpdateNearestTocCore(IHostService host, string item, FileModel toc, Dictionary<string, Toc> nearest)
+        {
+            var allSourceFiles = host.SourceFiles;
+            var tocOutputFile = GetOutputPath(toc.FileAndType);
+            FileAndType itemSource;
+            if (allSourceFiles.TryGetValue(item, out itemSource))
+            {
+                var itemOutputFile = GetOutputPath(itemSource);
+                var relative = tocOutputFile.RemoveWorkingFolder() - itemOutputFile;
+                Toc cur;
+                lock (nearest)
+                {
+                    if (!nearest.TryGetValue(item, out cur) || CompareRelativePath(relative, cur.OutputPath) < 0)
```

Comment:
got it. TOC will always be built.

---
In reply to: [100466512](https://github.com/dotnet/docfx/pull/1285#discussion_r100466512) [](ancestors = 100466512,100465411)